src/main/java/se/liu/ida/hefquin/cli/modules/ModFederation.java

```diff
@@ -105,7 +103,12 @@ else if ( addr.isURIResource() ) {
 				throw new IllegalArgumentException();
 			}
 
-			addSPARQLEndpoint(addrStr);
+			if( fd.contains(iface, FD.vocabularyMappings) ){
+				final RDFNode path = fd.getRequiredProperty(iface, FD.vocabularyMappings).getObject();
+				addSPARQLEndpoint(addrStr, path.toString());
```

Comment:
Why have that only for SPARQL endpoints and not for any other type of federation member? I think that should be general. To achieve that--in combination with my other comment (about making `fd:vocabularyMappingsFile` a property of the fed.member)--you will probably have to wrap all these loops here into another loop that iterates over all fed.members. Then, you can process the `fd:vocabularyMappingsFile` property first and create a `VocabularyMapping` object from the file (and have that `VocabularyMapping` object be `null` for all the fed.members that do not have a `fd:vocabularyMappingsFile` property).

Another issue: you should explicitly check whether there is indeed a file at the given path.